Mathlib/Analysis/Complex/IntegerCompl.lean

```diff
@@ -0,0 +1,55 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Birkbeck
+-/
+
+import Mathlib.Analysis.Complex.UpperHalfPlane.Basic
+
+/-!
+# Integer Complement
+
+We define the complement of the integers in the complex plane and give some basic lemmas about it.
+We also show that the upper half plane embeds into the integer complement.
+
+-/
+
+open UpperHalfPlane
+
+/--The complement of the integers in `ℂ`. -/
+abbrev Complex.IntegerComplement := (Set.range ((↑) : ℤ → ℂ))ᶜ
+
+namespace Complex
+
+local notation "ℂ_ℤ " => IntegerComplement
+
+lemma IntegerComplement_eq : ℂ_ℤ = {z : ℂ | ¬ ∃ (n : ℤ), z = ↑n} := by
+  rw [IntegerComplement]
+  aesop
+
+lemma IntegerComplement_not_exist {x : ℂ} (hx : x ∈ ℂ_ℤ) : ¬ ∃ (n : ℤ), x = ↑n := by
+  rw [IntegerComplement_eq] at hx
+  exact hx
+
+lemma IntegerComplement_mk {x : ℂ} (hx : ¬ ∃ (n : ℤ), x = ↑n) : x ∈ ℂ_ℤ := by
+  rw [IntegerComplement_eq]
+  exact hx
```

Comment:
Likewise:
```suggestion
lemma IntegerComplement_not_exist {x : ℂ} (hx : x ∈ ℂ_ℤ) : ¬ ∃ (n : ℤ), n = x := hx

lemma IntegerComplement_mk {x : ℂ} (hx : ¬ ∃ (n : ℤ), n = x) : x ∈ ℂ_ℤ := hx
```
But actually, the latter is misnamed now that it's not a subtype. I would probably just turn this into a `rw` lemma like this:
```
lemma integerComplement.mem_iff {x : ℂ} : x ∈ ℂ_ℤ ↔ ¬ ∃ (n : ℤ), n = x := Iff.rfl
```
and then you can add `alias`es if you prefer.